src/main/java/thePackmaster/hats/HatMenu.java

```diff
@@ -82,7 +88,12 @@ public static void refreshHatDropdown() {
         dropdown = new DropdownMenu(((dropdownMenu, index, s) -> setCurrentHat(index, s)),
                 optionNames, FontHelper.tipBodyFont, Settings.CREAM_COLOR);
 
-        if (init) setCurrentHat(0, optionNames.get(0));
+
+        if (init) {
+            int lastPickedIdx = SpireAnniversary5Mod.getLastPickedHatIndex();
```

Comment:
This is going to be unstable -- whenever we add another pack, indexes in the list will change.

What I suggest we do here is:
* Track the mapping of pack ID to dropdown index (in this class, when building the dropdown)
* Save the pack ID, not the index
* Have this code using the mapping to look up the right index

Reply:
hmm - I started on this, but that actually doesn't work with the Randomized Hat option right now since that one isn't attached to an ID in the hats and only exists in the options

Reply:
OK, got it